pmda/density.py

```diff
@@ -0,0 +1,222 @@
+            any value that evaluates to "False" (such as the default 0) disables this
+            special selection.
+    update_selection : bool (optional)
+            Should the selection of atoms be updated for every step? ["False"]
+            - "True": atom selection is updated for each frame, can be slow
+            - "False": atoms are only selected at the beginning
+    verbose : bool (optional)
+            Print status update to the screen for every *interval* frame? ["True"]
+            - "False": no status updates when a new frame is processed
+            - "True": status update every frame (including number of atoms
+              processed, which is interesting with "update_selection=True")
+    interval : int (optional)
+           Show status update every `interval` frame [1]
+    parameters : dict (optional)
+            `dict` with some special parameters for :class:`Density` (see docs)
+    gridcenter : numpy ndarray, float32 (optional)
+            3 element numpy array detailing the x, y and z coordinates of the
+            center of a user defined grid box in Angstroem ["None"]
+    xdim : float (optional)
+            User defined x dimension box edge in ångström; ignored if
+            gridcenter is "None"
+    ydim : float (optional)
+            User defined y dimension box edge in ångström; ignored if
+            gridcenter is "None"
+    zdim : float (optional)
+            User defined z dimension box edge in ångström; ignored if
+            gridcenter is "None"
+
+    Returns
+    -------
+    :class:`Density`
+            A :class:`Density` instance with the histogrammed data together
+            with associated metadata.
+    """"
+
+    def __init__(self, atomgroup, delta=1.0, atomselection="name OH2", start=None,
+                stop=None, step=None, metadata=None, padding=2.0, cutoff=0,
+                soluteselection=None, use_kdtree=True, update_selection=False,
+                verbose=False, interval=1, quiet=None, parameters=None,
+                gridcenter=None, xdim=None, ydim=None, zdim=None):
+        """
+        Parameters
+        ----------
+        atomgroup : AtomGroup
+            atom groups that are iterated over in parallel
+        """
+        u = atomgroup.universe
+        super(pDensity, self).__init__(u, (atomgroup, ))
+        self._atomgroup = atomgroup
+        self._trajectory = u.trajectory
+        self._cutoff = cutoff
+        self._atomselection = atomselection
+        self._soluteselection = soluteselection
+        self._update_selection = update_selection
+        self._padding = padding
+        self._metadata = metadata
+        self._parameters = parameters
+        self._n_frames = u.trajectory.n_frames
+        coord = self.current_coordinates(atomgroup, cutoff, atomselection, soluteselection, update_selection)
+        box, angles = self._trajectory.ts.dimensions[:3], self._trajectory.ts.dimensions[3:]
+        # logger.info(
```

Comment:
remove logger (not used elsewhere in PMDA – or do we???)

Reply:
We don't use logger elesewhere in PMDA.